lib/modules/versioning/rust-release-channel/index.ts

```diff
@@ -0,0 +1,403 @@
+import type { NewValueConfig, VersioningApi } from '../types';
+import { parse } from './parse';
+import type { ToolchainObject } from './types';
+
+export const id = 'rust-release-channel';
+export const displayName = 'Rust Release Channel';
+export const urls = [
+  'https://rust-lang.github.io/rustup/concepts/toolchains.html',
+];
+export const supportsRanges = true;
+export const supportedRangeStrategies = ['pin', 'replace'];
+
+const rust1Date = new Date('2015-05-15');
+
+class RustReleaseChannelVersioning implements VersioningApi {
+  isValid(input: string): boolean {
+    return parse(input) !== null;
+  }
+
+  isVersion(input: string | undefined | null): boolean {
+    if (!input) {
```

Comment:
After importing `isNonEmptyString from @sindresorhus/is`
```suggestion
    if (!isNonEmptyString(input)) {
```